support/timepb/cmp.go

```diff
@@ -0,0 +1,92 @@
+package timepb
+
+import (
+	"fmt"
+	"time"
+
+	durpb "google.golang.org/protobuf/types/known/durationpb"
+	tspb "google.golang.org/protobuf/types/known/timestamppb"
+)
+
+// IsZero returns true when t is nil or is zero unix timestamp (1970-01-01)
+func IsZero(t *tspb.Timestamp) bool {
+	return t == nil || t.Nanos == 0 && t.Seconds == 0
+}
+
+// Commpare t1 and t2 and returns -1 when t1 < t2, 0 when t1 == t2 and 1 otherwise.
+// Returns false if t1 or t2 is nil
+func Compare(t1, t2 *tspb.Timestamp) int {
+	if t1 == nil || t2 == nil {
+		panic(fmt.Sprint("Can't compare nil time, t1=", t1, "t2=", t2))
+	}
+	if t1.Seconds == t2.Seconds && t1.Nanos == t2.Nanos {
+		return 0
+	}
+	if t1.Seconds < t2.Seconds || t1.Seconds == t2.Seconds && t1.Nanos < t2.Nanos {
+		return -1
+	}
+	return 1
+}
+
+// DurationIsNegative returns true if the duration is negative. It assumes that d is valid
+// (d..CheckValid() is nil).
+func DurationIsNegative(d *durpb.Duration) bool {
+	return d.Seconds < 0 || d.Seconds == 0 && d.Nanos < 0
+}
+
+// AddStd returns a new timestamp with value t + d, where d is stdlib Duration.
+// If t is nil then nil is returned.
+// Panics on overflow.
+func AddStd(t *tspb.Timestamp, d time.Duration) *tspb.Timestamp {
```

Comment:
I think i would prefer to have this called `AddDuration` or `AddStdDuration`,

Reply:
In stdlib we have `time.Add` not `time.AddDuration`. You can't add time to another time, it doesn't make sense.

Reply:
I think the short naming is okay